picasso/src/main/java/com/squareup/picasso/RequestCreator.java

```diff
@@ -373,8 +378,19 @@ public void fetch() {
 
       Request request = createRequest(started);
       String key = createKey(request, new StringBuilder());
+      Bitmap bitmap = picasso.quickMemoryCacheCheck(key);
```

Comment:
fetch() is allowed to be invoked by any other thread other than the main thread. This assumes the cache implementation is thread safe...perhaps it was intentional we never added this here.